### PR TITLE
Cache github API calls to disk

### PR DESCRIPTION
**Pull request by YenTheFirst**
Addresses [https://github.com/telia-oss/github-pr-resource/issues/142]

When this resource is configured to filter versions, it can sometimes
result in a large number of repeated calls to the github API to
determine whether a PR should be emitted as a version or not.
In extreme cases, this can result in usage that exceeds Github's API
rate limit, causing the resource to be unavailable until the rate limit
resets.

This change mitigates the issue by caching the results of Github API
calls to disk, such that repeated calls to fetch the modified file list
of Pull Requests which haven't updated since the previous call, will
instead be served from the cache.

As Concourse intentionally re-uses containers for Check processes over
a long time period, this cache should be long-lived in most deployments.
[https://concourse-ci.org/implementing-resource-types.html#resource-check]

It should be noted that writing a cached response to disk can fail.
The underlying library, peterbourgon/diskv, returns a result in these
cases. However, the http caching library, gregjones/httpcache, silently
swallows any such errors. If a permission issue continually prevents
writing cached responses to disk, the Check will still proceed as normal,
and this resource will behave as if caching wasn't implemented.

### CODE REVIEW

File: github.go [hunk 1]
```diff
@@ -13,10 +13,18 @@ import (
...
+	"github.com/gregjones/httpcache/diskcache"
 	"github.com/shurcooL/githubv4"
 	"golang.org/x/oauth2"
 )
 
+// on-disk location to store cached API results in between invocations
+// primarily useful for the Check command, which will be re-run in the same container multiple times
+// the underlying disk caching library automatically finds-or-creates this directory.
+// if the given path can't be created as a writeable directory, the whole thing will silently fail to cache.
+const diskCachePath = "/tmp/github-api-cache"
```
Comment (lc-od) [inline github.go]: Use https://golang.org/pkg/io/ioutil/#TempDir?

Comment (YenTheFirst) [reply github.go]: in the case of `ioutil.TempDir` - this function intentionally creates a randomly named directory, intended to be globally unique. This means that two runs won't use the same cache directory, bypassing the point.

TempDir basically returns `/tmp`, which maybe makes constructing a shared cache path a bit more reasonable in a cross-platform way, but there's no guarantee that the directory exists or is writeable.

Comment (lc-od) [reply github.go]: `os.TempDir` should be OS agnostic.  I would assume the check container runs in privilege mode?

**Comment by YenTheFirst** [comment]
The upstream issue has not been resolved, but it doesn't make sense for me to hold open this PR on this fork. Feel free to re-open.

**Pull request closed without merging**